lib/sensors/bms.hpp

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <array>
+#include <optional>
+
+#include <bms_processor.hpp>
+#include <core/logger.hpp>
+#include <io/hardware_can.hpp>
+
+namespace hyped::sensors {
+
+// TODOLater: Confirm with propulsion
+static constexpr std::uint8_t NUM_BMS_CELLS               = 12;
+static constexpr std::uint8_t NUM_BMS_TEMPERATURE_SENSORS = 4;
```

Comment:
style guide?

Reply:
'tis but a draft, the whole PR was  for discussing structure. Once its ready I'll open it for reviews.